File: fix_zod.js
const fs = require('fs');

const files = [
    'f:\\Antigravity\\client\\src\\pages\\team\\StaffFormModal.tsx',
    'f:\\Antigravity\\client\\src\\pages\\team\\OnboardingPage.tsx'
];

for (const file of files) {
    let content = fs.readFileSync(file, 'utf8');

    // Replace .optional().or(z.literal('')) with .nullish().or(z.literal(''))
    content = content.replace(/\.optional\(\)\.or\(z\.literal\(''\)\)/g, ".nullish().or(z.literal(''))");

    // Replace z.string().optional() with z.string().nullish().or(z.literal(''))
    content = content.replace(/z\.string\(\)\.optional\(\)/g, "z.string().nullish().or(z.literal(''))");

    // Replace z.enum([...]).optional() with .nullish().or(z.literal(''))
    content = content.replace(/z\.enum\(\[\s*(?:'[^']*'(?:\s*,\s*)?)*\s*\]\)\.optional\(\)/g, match => match.replace('.optional()', ".nullish().or(z.literal(''))"));

    // Replace z.coerce.number().optional() with z.coerce.number().nullish()
    content = content.replace(/z\.coerce\.number\(\)\.optional\(\)/g, "z.coerce.number().nullish()");

    // Any surviving .optional() which might cause issues (safeguard)
    content = content.replace(/\.nullable\(\)\.or\(z\.literal\(''\)\)/g, ".nullish().or(z.literal(''))");

    fs.writeFileSync(file, content, 'utf8');
}
console.log("Fix completed.");

File: fix_zod.py
import re

files = [
    r'f:\Antigravity\client\src\pages\team\StaffFormModal.tsx',
    r'f:\Antigravity\client\src\pages\team\OnboardingPage.tsx'
]

for filepath in files:
    with open(filepath, 'r') as f:
        content = f.read()

    # Replace .optional().or(z.literal('')) with .nullish().or(z.literal(''))
    content = content.replace(".optional().or(z.literal(''))", ".nullish().or(z.literal(''))")
    
    # Replace remaining .optional() on z.string() with .nullish().or(z.literal(''))
    content = content.replace("z.string().optional()", "z.string().nullish().or(z.literal(''))")

    # Replace .optional() on z.enum(...) with .nullish()
    content = re.sub(r'z\.enum\(\[.*?\]\)\.optional\(\)', lambda m: m.group(0).replace('.optional()', ".nullish().or(z.literal(''))"), content)

    # For z.coerce.number().optional()
    content = content.replace("z.coerce.number().optional()", "z.coerce.number().nullish()")

    with open(filepath, 'w') as f:
        f.write(content)

print("Fix completed.")
